hepdata/modules/records/utils/submission.py

```diff
@@ -292,15 +315,19 @@ def process_general_submission_info(basepath, submission_info_document, recid):
     if "modifications" in submission_info_document:
         parse_modifications(hepsubmission, recid, submission_info_document)
 
-    if 'additional_resources' in submission_info_document:
-
-        for reference in hepsubmission.resources:
-            db.session.delete(reference)
+    cleanup_data_resources(hepsubmission)
 
+    if 'additional_resources' in submission_info_document:
         resources = parse_additional_resources(basepath, recid, submission_info_document)
         for resource in resources:
             hepsubmission.resources.append(resource)
 
+    if hepsubmission.overall_status not in ("sandbox", "sandbox_processing"):
+        if 'related_to_hepdata_recids' in submission_info_document:
+            for related_id in submission_info_document['related_to_hepdata_recids']:
```

Comment:
Now change `related_to_hepdata_recids` to `related_to_hepdata_records` to match latest `hepdata-validator` release v0.3.4.  Other variables like the database fields do not need to be changed.